tests/integration/multinodes.nim

```diff
@@ -311,7 +311,7 @@ template multinodesuite*(name: string, body: untyped) =
                           role: Role.Client,
                           node: node
                         )
-            if clients().len == 1:
+            if running.len == 1:
```

Comment:
This is behind `if var clients =? nodeConfigs.clients:` and nodeConfigs.clients is of type `?CodexConfigs`. So these should always be Codexes. I changed this because using `clients()` assumes that the config for this test has at least 1 client configured and that the client will always be started before any providers/validators (true because of the ordering of the code). If a user creates a config with only providers or validators, none of them would be used for bootstrap. With this change, the first node regardless of role will be the bootstrap node.

Reply:
If a user creates a configuration that runs a single hardhat node, `running.len == 1` will be true when processing the first client node. We don't want that.

Reply:
> If a user creates a config with only providers or validators, none of them would be used for bootstrap. With this change, the first node regardless of role will be the bootstrap node.

Without a client config, there will be no bootstrap nodes, since this is nested under the iteration of `clients.configs`. 

I've made some changes that add each started client and validator to a seq of bootstrap nodes, that are used to start the next node. That means each subsequently started node should be more connected to peers than the last. This also fixes the case when no clients were started, that bootstrap SPRs are still used. PR https://github.com/codex-storage/nim-codex/pull/1048